src/main/java/com/redhat/devtools/lsp4ij/settings/jsonSchema/LSPServerConfigurationJsonSchemaFileProvider.java

```diff
@@ -0,0 +1,84 @@
+package com.redhat.devtools.lsp4ij.settings.jsonSchema;
+
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.impl.PsiManagerEx;
+import com.intellij.psi.impl.file.impl.FileManagerImpl;
+import com.intellij.util.ModalityUiUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+public class LSPServerConfigurationJsonSchemaFileProvider extends AbstractLSPJsonSchemaFileProvider {
+
+    private static final Object SCHEMA_CHANGE_REQUESTOR = new Object();
+
+    private final int index;
+    private final @NotNull Project project;
+    private MyLightVirtualFile file;
+    private boolean unused;
+
+    public LSPServerConfigurationJsonSchemaFileProvider(int index, @NotNull Project project) {

```

Comment:
Note to self: What is `index` here?

Reply:
add a comment to explain what is the index.